xarray/tests/test_backends.py

```diff
@@ -570,6 +585,20 @@ def create_tmp_file(suffix='.nc', allow_cleanup_failure=False):
 
 
 @contextlib.contextmanager
+def create_tmp_file_pathlib(suffix='.nc', allow_cleanup_failure=False):
```

Comment:
Rather than repeating everything from `create_tmp_file`, can you simply wrap the output from `create_tmp_file` in `Path`?